src/plugins/intel_cpu/src/transformations/cpu_opset/common/op/read_value_with_subgraph.cpp

```diff
@@ -0,0 +1,106 @@
+    return get_variable()->get_info().variable_id;
+}
+
+void ov::intel_cpu::ReadValueWithSubgraph::set_input(const Output<Node>& value,
+                                                         const std::shared_ptr<op::v0::Parameter>& body_parameter) {
+    OPENVINO_ASSERT(body_parameter != nullptr, "Missing parameter! parameter is is nullptr!");
+    auto param_index = m_bodies[0]->get_parameter_index(body_parameter);
+
+    OPENVINO_ASSERT(param_index != -1, "Missing parameter ", body_parameter->get_friendly_name(), " for \'body\'!");
+
+    set_invariant_inputs(value, {body_parameter});
+}
+
+ov::Output<ov::Node> ov::intel_cpu::ReadValueWithSubgraph::set_output(
+    const std::shared_ptr<op::v0::Result>& body_result) {
+    OPENVINO_ASSERT(body_result != nullptr, "Incorrect result in \"body\"! Result cant be \'nullptr\'");
+    auto result_id = m_bodies[0]->get_result_index(body_result);
+
+    OPENVINO_ASSERT(result_id != -1, "Missing result ", body_result->get_friendly_name(), "in \'body\'!");
+
+    return set_body_outputs({body_result});
+}
+
+std::shared_ptr<ov::Node> ov::intel_cpu::ReadValueWithSubgraph::clone_with_new_inputs(
+    const OutputVector& new_args) const {
+    INTERNAL_OP_SCOPE(intel_cpu_ReadValueWithSubgraphNode_clone_with_new_inputs);
+
+    check_new_args_count(this, new_args);
+    auto op = std::make_shared<ov::intel_cpu::ReadValueWithSubgraph>();
+    OPENVINO_ASSERT(op.get(),
+                    op != nullptr,
+                    "Cannot clone ",
+                    description(),
+                    " operation with name ",
+                    get_friendly_name());
+    op->set_arguments(new_args);
+    op->set_output_size(m_output_descriptions[0].size());
+    op->set_function(get_function()->clone());
+    for (const auto& m_input_descr : m_input_descriptions[0]) {
+        op->m_input_descriptions[0].push_back(m_input_descr->copy());
+    }
+    for (const auto& m_output_descr : m_output_descriptions[0]) {
+        op->m_output_descriptions[0].push_back(m_output_descr->copy());
+    }
+    op->validate_and_infer_types();
+    return op;
+}
+
+bool ov::intel_cpu::ReadValueWithSubgraph::visit_attributes(AttributeVisitor& visitor) {
+    INTERNAL_OP_SCOPE(intel_cpu_ReadValueWithSubgraphNode_visit_attributes);
+    visitor.on_attribute("body", m_bodies[0]);
+    visitor.on_attribute("inputs", m_input_descriptions[0]);
+    visitor.on_attribute("outputs", m_output_descriptions[0]);
+    return true;
+}
+
+void ov::intel_cpu::ReadValueWithSubgraph::validate_and_infer_types() {
+    INTERNAL_OP_SCOPE(intel_cpu_ReadValueWithSubgraphNode_validate_and_infer_types);
+
+    NODE_VALIDATION_CHECK(this, m_bodies.size() == 1, "If contains incorrect number of bodies:", m_bodies.size());
+
+    NODE_VALIDATION_CHECK(this,
+                          m_input_descriptions.size() == 1,
+                          "If contains incorrect number of body input descriptions:",
```

Comment:
Probably the message has been taken from the `If` node. Please edit the message. The same is true for the outputs check.

Reply:
Done